sympy/solvers/solveset.py

```diff
@@ -978,6 +985,342 @@ def _solveset(f, symbol, domain, _check=False):
+    are invoked once the equation is identified by the `identifying
+    helpers`. These function return either the complete solution or
+    modifies the equation for `solveset` to handle.
+
+    * Philosophy behind the module
+
+    `\_transolve` comes into action when `solveset` is unable to solve
+    the equation as a last resort to get the solutions. The idea is to
+    get the equation in a case (Add, Pow etc.) that satisfies its
+    general form, for example exponential equations take the form as
+    `a*f(x) + b*g(x)`, where `f(x)` and `g(x)` are power terms,
+    therefore they are included in `Add` case.
+
+    Once done different `identifying helper` conditions are included
+    to check which class the equation belongs. `Solving helper` is
+    invoked once the class of the equation is identified. If the
+    equation is solved the result is returned otherwise a
+    `ConditionSet` is returned.
+
+    How `\_transolve` is better than `\_tsolve`
+    ===========================================
+
+    1) Better output
+
+    `\_transolve` provides a better output to some equations than
+    `\_tsolve`. Though result from both the functions are correct
+    it's just that the one from `\_transolve` is easy to understand and
+    appropriate. This is because `\_transolve` simplifies the result
+    wherever necessary.
+
+    Consider a simple exponential equation
+
+    >>> f = 3**(2*x) - 2**(x + 3)
+    >>> pprint(transolve(f, x, S.Reals), use_unicode=False)
+        -3*log(2)
+    {------------------}
+     -2*log(3) + log(2)
+    >>> pprint(tsolve(f, x), use_unicode=False)
+         /   3     \
+         | --------|
+         | log(2/9)|
+    [-log\2         /]
+
+    `\_transolve's` output seems more simple, readable and appropriate
+    as compared to that of `\_tsolve`.
+
+    2) Extensible
+
+    The API of `\_transolve` is designed such that it is easily extensible,
+    i.e, it is easy to add a new class of equation solver without having
+    to mess with the API or solving code of other helpers. The idea
+    behind `\_transolve` is making seperate helpers for different equation
+    solver. So to make new class of equation solve, you must add a
+    helper that identifies that particular class of the equation and
+    another helper that solves that type of equations.
+
+    3) Modular
+
+    `\_transolve` is designed to be modular i.e, for every class of
+    equation a separate helper for identification and solving is
+    implemented. This makes it easy to change or modify any of the
+    method implemented directly in the helpers without interfering
+    with the actual structure of the API.
+
+    4) Less complex API
+
+    The API of `\_transolve` is designed in a way that it is less
+    complex to read and understand unlike in `\_tsolve` where the
+    API structure is a mess with lots of recursive and function
+    calls making it hard to understand.
+
+    5) Faster Computation
```

Comment:
remove comparison (also, every class of equation that transolve attempts to identify will slow *it* down, too)